rosidl_generator_c/src/string_functions.c

```diff
@@ -25,10 +25,8 @@ rosidl_generator_c__String__init(rosidl_generator_c__String * str)
   if (!str) {
     return false;
   }
-  str->data = malloc(1);
-  if (!str->data) {
-    return false;
-  }
+  assert(str->data = malloc(1));
```

Comment:
Why assert here ? all assert lines will be removed if you build in release mode so I don't think that's what you want to do here